Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandRotateLogic.cs

```diff
@@ -0,0 +1,153 @@
+        /// a scalar applied to Rotation angle generated by handlebar calculations.
+        /// </summary>
+        private const float RotationMultiplier = 2f;
+
+        /// <summary>
+        /// This enum value stores the initial constraint specified as an argument in the Constructor.
+        /// It may be overridden by runtime conditions. The actual constraint is stored in the
+        /// variable currentRotationConstraint.
+        /// </summary>
+        private AxisConstraint rotationConstraint;
+
+        /// <summary>
+        /// Vector storing last handlebar Rotation. The handlebar is the line imagined between the controllers/hands.
+        /// </summary>
+        private Vector3 previousHandlebarRotation;
+
+        /// <summary>
+        /// The current rotation constraint might be modified based on disambiguation logic, for example
+        /// XOrYBasedOnInitialHandPosition might change the current rotation constraint based on the 
+        /// initial hand positions at the start
+        /// </summary>
+        private AxisConstraint currentRotationConstraint;
+
+        /// <summary>
+        /// ProjectHandlebarGivenConstraint internal function to account for axis constraint
+        /// </summary>
+        /// <param name="constraint">Enum value describing the axis to which the rotation is constrained/param>
+        /// <param name="handlebarRotation">A Vector3 describing the rotation of the line connecting the inputSources</param>
+        /// <param name="manipulationRoot">Transform of gameObject to be two hand manipulated</param>
+        /// <returns>a Vector3 describing handlebar after constraint is applied</returns>
+        private Vector3 ProjectHandlebarGivenConstraint(AxisConstraint constraint, Vector3 handlebarRotation, Transform manipulationRoot)
+        {
+            Vector3 result = handlebarRotation;
+            switch (constraint)
+            {
+                case AxisConstraint.XAxisOnly:
+                    result.x = 0;
+                    break;
+                case AxisConstraint.YAxisOnly:
+                    result.y = 0;
+                    break;
+                case AxisConstraint.ZAxisOnly:
+                    result.z = 0;
+                    break;
+            }
+            return CameraCache.Main.transform.TransformDirection(result);
+        }
+
+        /// <summary>
+        /// GetHandlebarDirection internal function to get rotation described by inputSources.
+        /// </summary>
+        /// <param name="handsPressedMap">Dictionary listing inputSourceStates</param>
+        /// <param name="manipulationRoot">Transform of gameObject to be two hand manipulated</param>
+        /// <returns>A Vector3 describing the direction of the line connecting the inputSources</returns>
+        private Vector3 GetHandlebarDirection(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot)
+        {
+            Assert.IsTrue(handsPressedMap.Count > 1);
```

Comment:
Let's use `UnityEngine.Debug.Asset` instead.  This is mainly used for Unit Tests.